src/index.js

```diff
@@ -173,6 +173,19 @@ function EvolvClient(options) {
    */
   this.emit = function(type, metadata, flush) {
     context.pushToArray('events', {type: type,  timestamp: (new Date()).getTime()});
+
+    context.set('fired_events.'+type, true, true);
+    const firedEvents = sessionStorage.getItem('fe');
+    let eventsList = {};
+    if(firedEvents) {
+      const firedEventsObj = JSON.parse(firedEvents);
+      firedEventsObj[type] = true;
+      eventsList= firedEventsObj;
+    } else {
+      eventsList[type] = true;
+    }
+    sessionStorage.setItem('fe', JSON.stringify(eventsList));
+
```

Comment:
The `javascript-sdk` is supposed to be runtime agnostic; it could theoretically run under Node as well as the browser. This is the reason that `javascript-sdk` and `asset-manager` are broken out into separate packages.

Therefore it won't have access to `sessionStorage`. At the moment I don't know where the right place to store values would be instead. Perhaps @tfoster has a suggestion

Reply:
Good point, I didn't know that.